pkg/telemetry/tracing.go

```diff
@@ -1,12 +1,51 @@
 package telemetry
 
 import (
+	"context"
+
+	"github.com/openfga/openfga/internal/build"
+	"go.opentelemetry.io/otel"
 	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.12.0"
 	"go.opentelemetry.io/otel/trace"
+	"google.golang.org/grpc"
 )
 
-func NewNoopTracer() trace.Tracer {
-	return trace.NewNoopTracerProvider().Tracer("noop")
+func MustNewTracerProvider(endpoint string, ratio float64) *sdktrace.TracerProvider {
+	res, err := resource.Merge(
+		resource.Default(),
+		resource.NewSchemaless(
+			semconv.ServiceNameKey.String("openfga"),
+			semconv.ServiceVersionKey.String(build.Version),
+		))
+	if err != nil {
+		panic(err)
+	}
+
+	exp, err := otlptracegrpc.New(context.Background(),
```

Comment:
This makes sense. Hopefully we won't need to add more in the future because everyone will just use the grpc endpoint. For the moment, how about we just rename the flag to `trace-otlgrpc-endpoint` and if we do need to add more in the future we can without breaking anything?

Reply:
I think that's a good plan 👍